osu.Game/Online/DownloadTrackingComposite.cs

```diff
@@ -46,25 +47,41 @@ private void load()
             {
                 if (modelInfo.NewValue == null)
                     attachDownload(null);
-                else if (manager?.IsAvailableLocally(modelInfo.NewValue) == true)
+                else if (IsModelAvailableLocally())
                     State.Value = DownloadState.LocallyAvailable;
                 else
-                    attachDownload(manager?.GetExistingDownload(modelInfo.NewValue));
+                    attachDownload(Manager?.GetExistingDownload(modelInfo.NewValue));
             }, true);
 
-            if (manager == null)
+            if (Manager == null)
                 return;
 
-            managerDownloadBegan = manager.DownloadBegan.GetBoundCopy();
+            managerDownloadBegan = Manager.DownloadBegan.GetBoundCopy();
             managerDownloadBegan.BindValueChanged(downloadBegan);
-            managerDownloadFailed = manager.DownloadFailed.GetBoundCopy();
+            managerDownloadFailed = Manager.DownloadFailed.GetBoundCopy();
             managerDownloadFailed.BindValueChanged(downloadFailed);
-            managedUpdated = manager.ItemUpdated.GetBoundCopy();
+            managedUpdated = Manager.ItemUpdated.GetBoundCopy();
             managedUpdated.BindValueChanged(itemUpdated);
-            managerRemoved = manager.ItemRemoved.GetBoundCopy();
+            managerRemoved = Manager.ItemRemoved.GetBoundCopy();
             managerRemoved.BindValueChanged(itemRemoved);
         }
 
+        /// <summary>
+        /// Verifies that the given databased model is in a correct state to be considered available.
+        /// </summary>
+        /// <example>
+        /// In the case of multiplayer/playlists, this has to verify that the databased beatmap set with the selected beatmap matches what's online.
```

Comment:
This is broken (not sure how to read or what this function does).

Reply:
Yeah as usual, still bad at documenting since day 1, what about this then:
```
In the case of multiplayer/playlists, this method will be overriden to verify that the imported beatmap set with the beatmap selected in room matches what's online.
```
or maybe remove the "matches what's online" and directly say "has the correct checksum set."

Reply:
I'd suggest something like:
```csharp
/// <summary>
/// Checks that a database model matches the one expected to be downloaded.
/// </summary>
```